models/receipt.go

```diff
@@ -22,26 +22,68 @@ import (
 // geth node has the data locally, but we don't in evm gateway, so we can not reproduce those values
 // and we need to store them
 type StorageReceipt struct {
-	Type              uint8
-	PostState         []byte
-	Status            uint64
-	CumulativeGasUsed uint64
+	Type              uint8  `json:"type,omitempty"`
+	PostState         []byte `json:"root"`
+	Status            uint64 `json:"status"`
+	CumulativeGasUsed uint64 `json:"cumulativeGasUsed"`
 	// todo we could skip bloom to optimize storage and dynamically recalculate it
-	Bloom             gethTypes.Bloom
-	Logs              []*gethTypes.Log
-	TxHash            common.Hash
-	ContractAddress   common.Address
-	GasUsed           uint64
-	EffectiveGasPrice *big.Int
-	BlobGasUsed       uint64
-	BlobGasPrice      *big.Int
-	BlockHash         common.Hash
-	BlockNumber       *big.Int
-	TransactionIndex  uint
+	Bloom             gethTypes.Bloom  `json:"logsBloom"`
+	Logs              []*gethTypes.Log `json:"logs"`
+	TxHash            common.Hash      `json:"transactionHash"`
+	ContractAddress   common.Address   `json:"contractAddress"`
+	GasUsed           uint64           `json:"gasUsed"`
+	EffectiveGasPrice *big.Int         `json:"effectiveGasPrice"`
+	BlobGasUsed       uint64           `json:"blobGasUsed,omitempty"`
+	BlobGasPrice      *big.Int         `json:"blobGasPrice,omitempty"`
+	BlockHash         common.Hash      `json:"blockHash,omitempty"`
+	BlockNumber       *big.Int         `json:"blockNumber,omitempty"`
+	TransactionIndex  uint             `json:"transactionIndex"`
+	RevertReason      []byte           `json:"revertReason"`
```

Comment:
Couldn't we make this just be:

```
type StorageReceipt struct {
    *gethTypes.Receipt
    RevertReason      []byte
}
```

Reply:
Unfortunately this does not work, because the RLP encoding of a `gethTypes.Receipt` includes only consensus fields. See: https://github.com/onflow/go-ethereum/blob/master/core/types/receipt.go#L120-L123
And in our implementation, we have decided to store all the fields, that's why we copied the fields in the `StorageReceipt` in the first place.

Reply:
Ohh I remember now. Could we maybe define our own RLP encode call on this type so it includes more fields and this way we don't have to define the fields.

Reply:
That's a good idea :+1: 
I need to try that out and get back to you.

Reply:
> Ohh I remember now. Could we maybe define our own RLP encode call on this type so it includes more fields and this way we don't have to define the fields.

@sideninja  I had a look at this. Even if we define `EncodeRLP` & `DecodeRLP` on `StorageReceipt`, one of the fields is of type `*gethTypes.Receipt`, so there's no way to override how the RLP encoding/decoding for that field is performed. It will always call `*gethTypes.Receipt.EncodeRLP()`, which includes only the few consensus fields. I guess there's nothing to be done about it, other than include all of its fields on `StorageReceipt`.